internal/storage/v2/clickhouse/tracestore/dbmodel/to_dbmodel.go

```diff
@@ -0,0 +1,422 @@
+// Therefore, splitting the key and value of pcommon.Map into arrays for storage ensures that 1. data types are not lost
+// and 2. it can be more conveniently used as query parameters.
+type AttributesGroup struct {
+	BoolKeys     []string
+	BoolValues   []bool
+	DoubleKeys   []string
+	DoubleValues []float64
+	IntKeys      []string
+	IntValues    []int64
+	StrKeys      []string
+	StrValues    []string
+	BytesKeys    []string
+	BytesValues  []string
+}
+
+// AttributesToGroup Categorizes and aggregates Attributes based on the data type of their values, and writes them in batches.
+func AttributesToGroup(attributes pcommon.Map) AttributesGroup {
+	attributesMap := AttributesToMap(attributes)
+	var group AttributesGroup
+	for valueType := range attributesMap {
+		kvPairs := attributesMap[valueType]
+		switch valueType {
+		case ValueTypeBool:
+			for k, v := range kvPairs {
+				group.BoolKeys = append(group.BoolKeys, k)
+				group.BoolValues = append(group.BoolValues, v.Bool())
+			}
+		case ValueTypeDouble:
+			for k, v := range kvPairs {
+				group.DoubleKeys = append(group.DoubleKeys, k)
+				group.DoubleValues = append(group.DoubleValues, v.Double())
+			}
+		case ValueTypeInt:
+			for k, v := range kvPairs {
+				group.IntKeys = append(group.IntKeys, k)
+				group.IntValues = append(group.IntValues, v.Int())
+			}
+		case ValueTypeStr:
+			for k, v := range kvPairs {
+				group.StrKeys = append(group.StrKeys, k)
+				group.StrValues = append(group.StrValues, v.Str())
+			}
+		case ValueTypeBytes:
+			for k, v := range kvPairs {
+				group.BytesKeys = append(group.BytesKeys, k)
+				byteStr := base64.StdEncoding.EncodeToString(v.Bytes().AsRaw())
+				group.BytesValues = append(group.BytesValues, byteStr)
+			}
+		default:
+		}
+	}
+	return group
+}
+
+// AttributesToMap Groups a pcommon.Map by data type and splits the key-value pairs into arrays for storage.
+// The values in the key-value pairs of a pcommon.Map instance may not all be of the same data type.
+// For example, a pcommon.Map can contain key-value pairs such as:
+// string-string, string-bool, string-int64, string-float64. Clearly, the key-value pairs need to be classified based on the data type.
+func AttributesToMap(attrs pcommon.Map) map[pcommon.ValueType]map[string]pcommon.Value {
+	result := make(map[pcommon.ValueType]map[string]pcommon.Value)
+	for _, valueType := range []pcommon.ValueType{
+		ValueTypeBool, ValueTypeDouble, ValueTypeInt, ValueTypeStr, ValueTypeBytes,
+	} {
+		result[valueType] = make(map[string]pcommon.Value)
+	}
+	// Fill according to the data type of the value
+	for k, v := range attrs.All() {
+		typ := v.Type()
+		// For basic data types (such as bool, uint64, and float64) we can make sure type safe.
+		// TODO: For non-basic types (such as Map, Slice), they should be serialized and stored as OTLP/JSON strings
+		result[typ][k] = v
+	}
+	return result
+}
+
+var (
```

Comment:
this needs to be simplified, too much copy & paste and too easy to make a mistake.